### DISCUSSION TITLE
Build android node library in arm macOS

### DISCUSSION THREAD

**Discussion by Intro1997** [nodejs/help#4612] [Q&A]
I try to build android node library following [here](https://github.com/nodejs/node/blob/main/BUILDING.md#android) on my arm macbook. To get shared library, I modify the code in [android_configure.py:77](https://github.com/nodejs/node/blob/main/android_configure.py#L77) by adding `--shared`
```python
if os.path.exists("./configure"):
    os.system("./configure --dest-cpu=" + DEST_CPU + " --dest-os=android --openssl-no-asm --cross-compiling --shared")
```
but run `make -j4`  failed and I got following message:
```shell
ar crsT /Users/intro/Spaces/ThirdParty/node/out/Release/obj.target/deps/simdutf/libsimdutf.a @/Users/intro/Spaces/ThirdParty/node/out/Release/obj.target/deps/simdutf/libsimdutf.a.ar-file-list
ar: @/Users/intro/Spaces/ThirdParty/node/out/Release/obj.target/deps/simdutf/libsimdutf.a.ar-file-list: No such file or directory
make[1]: *** [/Users/intro/Spaces/ThirdParty/node/out/Release/obj.target/deps/simdutf/libsimdutf.a] Error 1
make[1]: *** Waiting for unfinished jobs....
rm 53810166e9269b4d0f73c5f6def0c7f6a6a25429.intermediate 163fe039f3c02ea66c2731d3c87d19d998b295dc.intermediate
make: *** [node] Error 2
```
Does it mean `ar` in macOS cannot use `@` symbol? And how can I solve it?

**Comment by Intro1997** [comment]
I solved this problem using GNU Binutils, but a new problem appeared, the error message is as follows:
```
/bin/sh: /Users/intro/Spaces/ThirdParty/node/out/Release/icupkg: cannot execute binary file
make[1]: *** [/Users/intro/Spaces/ThirdParty/node/out/Release/obj/gen/icudt75l.dat] Error 126
```
And I check the format of the file `icupkg`:
```
intro@introMacBook-Pro node % file /Users/intro/Spaces/ThirdParty/node/out/Release/icupkg
'/Users/intro/Spaces/ThirdParty/node/out/Release/icupkg: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /system/bin/linker64, not stripped
```
It's an ELF executable file and I cannot run it on macOS directly. Does it mean that I only can build android library on Linux?

**Reply by allenleeok** [reply]
Hi, I have the same problem, have you solved it?

**Reply by Intro1997** [reply]
> Hi, I have the same problem, have you solved it?

No, but I found that [this article](https://toyobayashi.github.io/2021/03/29/NodeEmbedding/) successfully built Android library with LTS v14.16.0 version, and the author provides library file in this [repo](https://github.com/toyobayashi/node-android-build). [v14.17.0](https://github.com/toyobayashi/node-android-build/releases/tag/v14.17.0) works fine for me.
